packages/grafana-data/src/utils/OptionsUIBuilders.ts

```diff
@@ -33,7 +32,7 @@ export class FieldConfigEditorBuilder<TOptions> extends OptionsUIRegistryBuilder
       override: standardEditorsRegistry.get('number').editor as any,
       editor: standardEditorsRegistry.get('number').editor as any,
       process: numberOverrideProcessor,
-      shouldApply: config.shouldApply ? config.shouldApply : (field) => field.type === FieldType.number,
+      shouldApply: config.shouldApply ?? (() => true),
```

Comment:
we had this strange default, where field options by default only applied to number fields. Removing this (potentially breaking change)